src/app/admin/product-form/product-form.component.ts

```diff
@@ -1,15 +1,29 @@
 import { Component, OnInit } from '@angular/core';
+import { AdminService } from './../admin.service';
+import { Category, Color } from './../product.model';
+import { Observable } from 'rxjs';
 
 @Component({
   selector: 'product-form',
   templateUrl: './product-form.component.html',
   styleUrls: ['./product-form.component.scss']
 })
 export class ProductFormComponent implements OnInit {
+  categoryList: Observable<Category[]>;
+  colors : string[]= [];
 
-  constructor() { }
+  constructor(private adminService: AdminService) {
+    this.categoryList = this.adminService.getCategories();
+    
+  }
 
   ngOnInit(): void {
+    this.colors = ['black', 'blue', 'green', 'pink', 'red', 'white', 'yellow']
```

Comment:
should not be used like this.. define at some common place

Reply:
iterate through the enum and get the keys